File: eternal_return/src/main/java/org/EternalReturn/ERCharacter/Character/adriana/LitFireOnAttack.java
package org.EternalReturn.ERCharacter.Character.adriana;

import org.EternalReturn.ERCharacter.ERCharacterMonobehaviour;
import org.EternalReturn.ERCharacter.Event.CharacterAttackEvent;
import org.EternalReturn.Util.dpengine.behaviour.MonobehaviourEvent;

import java.util.Collection;
import java.util.Objects;

public class LitFireOnAttack extends ERCharacterMonobehaviour<CharacterAttackEvent> {

    @Override
    public void start(CharacterAttackEvent event) {
        Objects.requireNonNull(event.victim.getEntity()).setFireTicks(100);
    }

    @Override
    public void update(Collection<MonobehaviourEvent> event) {

    }
}

File: eternal_return/src/main/kotlin/org/EternalReturn/ERCharacter/character/adriana/monobehavs/LitFireOnAttack.kt
package org.EternalReturn.ERCharacter.character.adriana.monobehavs

import org.EternalReturn.ERCharacter.ERCharacterMonobehaviour
import org.EternalReturn.ERCharacter.event.CharacterAttackEvent
import org.EternalReturn.util.dpengine.behaviour.MonobehaviourEvent
import org.bukkit.entity.Entity
import java.util.*

class LitFireOnAttack : ERCharacterMonobehaviour<CharacterAttackEvent>() {
    public override fun start(event: CharacterAttackEvent) {
        Objects.requireNonNull<Entity>(event.victim!!.entity).setFireTicks(100)
    }

    public override fun update(event: MutableCollection<MonobehaviourEvent>) {
    }
}
